### PR TITLE
refactor: Wifi station

### PR DESCRIPTION
**Pull request by abrondijk**
Brought `wifi::Station` to a fully functional and usable state, by actually handling events and by having implemented the new config implementation introduced by #32.

### CODE REVIEW

File: config_provider/include/ConfigProvider.hpp [hunk 1]
```diff
@@ -429,12 +432,21 @@ namespace sdk {
          * @brief Constructor to be used for incoming config changes
          * @param data Json object containing changes. Should only contain the fields that need to be updated
          */
-        explicit ConfigObject(nlohmann::json& data) {
+        explicit ConfigObject(const nlohmann::json& data) {
```
Comment (SquaredPotato) [inline config_provider/include/ConfigProvider.hpp:435]: Does this work when receiving json objects from the web server?

Comment (abrondijk) [reply config_provider/include/ConfigProvider.hpp:435]: Yes because we are not actually writing to this object

File: wifi/src/Station.cpp [hunk 2]
```diff
@@ -242,7 +242,7 @@ namespace sdk {
             m_restartType = RestartType::COMPONENT;
         }
 
-        etl::string<15> Station::getIpInfo() {
+        etl::string<15> Station::getAssignedIp() {
```
Comment (SquaredPotato) [inline wifi/src/Station.cpp:245]: If we support IPv6 we should make this string 45 characters long.

File: wifi/src/Station.cpp [hunk 3]
```diff
@@ -260,6 +260,50 @@ namespace sdk {
             return ip;
         }
 
+        etl::vector<Station::WifiRecord, 15> Station::scan() {
```
Comment (SquaredPotato) [inline wifi/src/Station.cpp]:
```suggestion
        etl::vector<Station::WifiRecord, CONFIG_AP_SCAN_NUMBER> Station::scan() {
```

File: wifi/src/Station.cpp [hunk 4]
```diff
@@ -260,6 +260,50 @@ namespace sdk {
...
+                ESP_LOGE(TAG, "Failed to get scan records: %s", esp_err_to_name(err));
+                return {};
+            }
+
+            if(scanCount > CONFIG_AP_SCAN_NUMBER){
+                ESP_LOGW(TAG, "Scanned more AP than the configured maximum, truncating to %d", CONFIG_AP_SCAN_NUMBER);
+                scanCount = CONFIG_AP_SCAN_NUMBER;
+            }
+
+            etl::vector<WifiRecord, 15> aps;
```
Comment (SquaredPotato) [inline wifi/src/Station.cpp]:
```suggestion
            etl::vector<WifiRecord, CONFIG_AP_SCAN_NUMBER> aps;
```

File: wifi/src/Station.cpp (hunk 2 above)
Comment (abrondijk) [reply wifi/src/Station.cpp:245]: Agreed

**Review by SquaredPotato — APPROVED** [review]

**Pull request merged**